subsys/bluetooth/host/hci_core.c

```diff
@@ -8237,7 +8237,10 @@ int bt_le_ext_adv_start(struct bt_le_ext_adv *adv,
 	atomic_set_bit_to(adv->flags, BT_ADV_LIMITED, param &&
 			  (param->timeout > 0 || param->num_events > 0));
 
-	le_adv_set_private_addr(adv);
+	if (IS_ENABLED(CONFIG_BT_PRIVACY) &&
+	    !atomic_test_bit(adv->flags, BT_ADV_USE_IDENTITY)) {
+		le_adv_set_private_addr(adv);
```

Comment:
@jhedberg @trond-snekvik  Any thoughts on how Mesh should be using this API and NRPA. Should every call to bt_le_ext_adv_start trigger a new NRPA set, or is it fine that a call to bt_le_ext_adv_update_param is required to do it?

Reply:
Is there a reason why ext adv should be different from legacy, where start() always generates a new NRPA?

Reply:
@jhedberg  Well, because the NRPA is kind of set from the parameters, which are in create() for ext adv.
Note that this will not set the NRPA in this version of the PR. So I would have to update it if that is the behavior we want.

Reply:
@joerchan as long as the code in `mesh/adv.c` doesn't get overly complex, I'm fine doing this whatever way you think makes most sense.

Reply:
Note that the current `start(); sleep(); stop()` method in `mesh/adv.c` will change to something asynchronous with a callback, since IIRC with ext adv you can state how many transmissions you want and then have a "sent" callback to notify of completion.

Reply:
Makes sense, so to me it looks like with the duration parameter then the mesh advertiser does not need to change the advertising interval, and the only API call needed would be adv_data_update and adv_start(duration=xmit).
I have updated the PR so that non-connectable advertiser will update the private advertiser when CONFIG_BT_PRIVACY=n
This should be similar behavior to legacy adv-start API.